common/presenters/framework-step-to-summary.js

```diff
@@ -1,15 +1,30 @@
-const { filter, flatten } = require('lodash')
+const { filter, find, flatten, isArray, isObject } = require('lodash')
 
 const frameworksHelpers = require('../helpers/frameworks')
 
 const frameworkFieldToSummaryListRow = require('./framework-field-summary-list-row')
 
 function frameworkStepToSummary(allFields, responses, baseUrl) {
-  return ([key, step]) => {
+  return ([key, step], i, allSteps = []) => {
+    const unmetStepConditions = allSteps
+      .map(step => step[1].next)
```

Comment:
I'm not sure if this is still possible (or relevant to this PR), but having an object instead of a tuple is more readable.
It took me a while to realise that `[1]` is the actual object that belongs to the step.

I think step should be `{url: string, step: { next: ..., slug: string, fields, string[]}}` instead of `[string, object]`

Reply:
I struggled a lot to get this neat too. Essentially what needs to happen is to remove any steps that are dependent on values within the responses. The only way to get it at this stage is to loop through `allSteps` which is all items in the map and search through the objects to find `next`. The tricky thing is that next can be a number of things too but for now we're only catering for basic conditional filtering.

I could convert it to an object but then you can't use native `map` and it would mean bringing in more lodash deps...I wonder if in that case this would make it more readable?

```javascript
.map(([key, step]) => step.next)
```

Reply:
> I could convert it to an object but then you can't use native map

You can always `Object.keys(obj).map(key => {})`?
But I agree, if the change is touching over components, it's probably wiser to defer it.